src/controllers/blocks/BlocksTraceController.ts

```diff
@@ -0,0 +1,66 @@
+import { ApiPromise } from '@polkadot/api';
+import { RequestHandler } from 'express-serve-static-core';
+
+import { BlocksTraceService } from '../../services';
+import AbstractController from '../AbstractController';
+import BlocksController from './BlocksController';
+
+export default class BlocksTraceController extends AbstractController<BlocksTraceService> {
+	constructor(api: ApiPromise) {
+		super(api, '/experimental/blocks', new BlocksTraceService(api));
+		this.initRoutes();
+	}
+
+	protected initRoutes(): void {
+		this.safeMountAsyncGetHandlers([
+			['/head/traces', this.getLatestBlockTraces],
+			['/:number/traces', this.getBlockTraces],
+			['/:number/traces/operations', this.getBlockOperations],
+			['/head/traces/operations', this.getLatestBlockOperations],
+		]);
+	}
+
+	private getLatestBlockTraces: RequestHandler = async (
+		_req,
+		res
+	): Promise<void> => {
```

Comment:
dq: how are the error propagated here? is it caught when the user calls `resolve/then` and `catch`?

I got a little confused with the return type `void`

Reply:
Yeah great question actually. I don't know if there is any idiomatic types for return errors with a neat enum like Rust. This service basically relies on all errors being thrown.

The caveat with express v4 `RequestHandler` functions is that they do not catch thrown errors in `async` functions. So we end up wrapping the call in `try { .. } catch { .. }` blocks. This effectively happens in `AbstrctController.safeMountAsyncGetHandlers`. Specifically it happens in [catchWrap](https://github.com/paritytech/substrate-api-sidecar/blob/01ae7ac1cbcb06cd76d6895bf9ccbe98e99f0d43/src/controllers/AbstractController.ts#L73-L90)